cmd/query/app/builder/storage.go

```diff
@@ -43,22 +42,21 @@ var (
 )
 
 // NewStorageBuilder creates a StorageBuilder based off the flags that have been set
-func NewStorageBuilder(opts ...basicB.Option) (StorageBuilder, error) {
-	flag.Parse()
+func NewStorageBuilder(parsedFlags *flags.Flags, opts ...basicB.Option) (StorageBuilder, error) {
```

Comment:
Could be possible, but see also: 
```
// TODO technically span and dependency storage might be separate
		return newCassandraBuilder(options.Cassandra, options.Logger, options.MetricsFactory, parsedFlags), nil
```
`newCassandraBuilder` requires `flags.DependencyStorage.DataFrequency`. What do you prefer? Add another method param or just reuse `SharedFlags`?